rskj-core/src/main/java/co/rsk/peg/PegUtils.java

```diff
@@ -0,0 +1,38 @@
+package co.rsk.peg;
+
+import co.rsk.bitcoinj.core.BtcTransaction;
+import co.rsk.bitcoinj.core.Sha256Hash;
+import co.rsk.bitcoinj.core.TransactionOutput;
+import co.rsk.bitcoinj.wallet.Wallet;
+import co.rsk.peg.bitcoin.BitcoinUtils;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.config.blockchain.upgrades.ConsensusRule;
+
+import java.util.List;
+import java.util.Optional;
+
+public class PegUtils {
+    private PegUtils() { }
+
+    public static PegTxType getTransactionType(
+        ActivationConfig.ForBlock activations,
+        BridgeStorageProvider provider,
+        Wallet liveFederationsWallet,
+        BtcTransaction btcTransaction
+    ) throws RegisterBtcTransactionException {
+        if (!activations.isActive(ConsensusRule.RSKIP379)){
+            throw new RegisterBtcTransactionException("Can't call PegUtil.getTransactionType before RSKIP379 activation.");
+        }
+
+        List<TransactionOutput> liveFederationOutputs = btcTransaction.getWalletOutputs(liveFederationsWallet);
```

Comment:
I think we discussed with Jose that we should first check if there is any value sent to any of the active federations, to avoid calculating the sighash (which may be an expensive operation) unnecessarily. If a random transaction is sent to the Bridge, we don't even bother in calculating the sighash. Which by the way we should a test for this case

Reply:
Is there a difference between using `getWalletOutputs` and `getValueSentToMe`? I think I have seen both being used

Reply:
Regarding your first comment, I forgot to mention to you that if we use that order and assume that when there are no outputs to any live fed, it is an UNKNOWN type. Pegouts with no change output will be missed because those txs will be caught in that condition, considering those pegouts as UNKNOWN.

This is not a big problem because no utxo is added to the available utxo lists, but that pegout won't be marked as processed, and we will be storing a useless sigHash for those pegouts.

Reply:
Regarding your second comment, yes, we also use `getValueSentToMe`. It is used for validating peg-in and flyover peg-ins. Here, I used `getWalletOutputs` since we are only interested in knowing if exists outputs for any live fed. The validation of the amount will be done later.